lib/spack/spack/audit.py

```diff
@@ -733,6 +733,26 @@ def _unknown_variants_in_dependencies(pkgs, error_cls):
         pkg_cls = spack.repo.PATH.get_pkg_class(pkg_name)
         filename = spack.repo.PATH.filename_for_package_name(pkg_name)
         for dependency_name, dependency_data in pkg_cls.dependencies.items():
+            # Check if there are nested dependencies declared. We don't want directives like:
+            #
+            #     depends_on('foo+bar ^fee+baz')
+            #
+            # but we'd like to have two dependencies listed instead.
+            for when, dependency_edge in dependency_data.items():
+                dependency_spec = dependency_edge.spec
+                nested_dependencies = dependency_spec.dependencies()
+                if nested_dependencies:
+                    summary = (
+                        f"{pkg_name}: invalid nested dependency "
+                        f"declaration '{str(dependency_spec)}'"
+                    )
+                    details = [
+                        f"split depends_on('{str(dependency_spec)}', when='{str(when)}') "
+                        f"into {len(nested_dependencies) + 1} directives",
+                        f"in {filename}",
+                    ]
+                    errors.append(error_cls(summary=summary, details=details))
```

Comment:
Given the name of the function and the docstring, I wouldn't have expected to find this check in it (the problem doesn't have to do with variants, and the dependency can be known) - I think updating the docstring would be helpful (IMO it could just list out the individual things that are checked for).

Also, does it make sense to put this before 

```
            if spack.repo.PATH.is_virtual(dependency_name):
                continue
```

? Virtuals can't have dependencies, so I'd assume we could also skip this check for virtuals in the same way we would skip the others.

Reply:
> Also, does it make sense to put this before ...

In my opinion yes, since this will catch issues with nested dependencies on virtuals - instead of silently dropping the dependency.

Reply:
Fixed in https://github.com/spack/spack/pull/41428/commits/e7c2cae13623e2d754c36702ff0a5fd1c1d34b45